security/cmd/istio_ca/main.go

```diff
@@ -187,6 +194,8 @@ func init() {
 		"The list of hostnames for istio ca server, separated by comma.")
 	flags.IntVar(&opts.grpcPort, "grpc-port", 8060, "The port number for Citadel GRPC server. "+
 		"If unspecified, Citadel will not serve GRPC requests.")
+	flags.BoolVar(&opts.serverOnly, "server-only", false, "When set, Citadel only serves as a server without writing "+
```

Comment:
seems both ```serverOnly==true``` and ```readSigningCertOnly==true``` mean current citadel instance only read k8s secret which is written by another instance ? If yes, could we just use one flag ?

Reply:
They are different, with serverOnly==true, Citadel won't write workload certs into secret; with readSigningCertOnly==true, Citadel doesn't generate signing certs (CA certs) if there isn't any. I think it's necessary to divide them, because in some cases users may want to control them separately. For example, clusters without secret volume mount, and use self-signed CAs, should use serverOnly=false for all, and readSigningCertOnly=true/false for different Citadels.